var/spack/repos/builtin/packages/py-torchgeo/package.py

```diff
@@ -35,113 +32,112 @@ class PyTorchgeo(PythonPackage):
     variant("tests", default=False, description="Install testing tools")
 
     # Required dependencies
-    depends_on("python@3.7:3+bz2", when="@0.3:", type=("build", "run"))
-    depends_on("python@3.6:3+bz2", when="@:0.2", type=("build", "run"))
-    depends_on("py-setuptools@42:67", when="@0.4.1:", type="build")
-    depends_on("py-setuptools@42:66", when="@0.4.0", type="build")
-    depends_on("py-setuptools@42:65", when="@0.3.1", type="build")
-    depends_on("py-setuptools@42:63", when="@:0.3.0", type="build")
-    depends_on("py-einops@0.3:0.6", when="@0.4:", type=("build", "run"))
-    depends_on("py-einops@0.3:0.4", when="@:0.3", type=("build", "run"))
-    depends_on("py-fiona@1.8:1", when="@0.3:", type=("build", "run"))
-    depends_on("py-fiona@1.5:1", when="@:0.2", type=("build", "run"))
-    depends_on("py-kornia@0.6.5:0.6", when="@0.4.1:", type=("build", "run"))
+    depends_on("python@3.9:", when="@0.5:", type=("build", "run"))
+    depends_on("python+bz2", type=("build", "run"))
+    depends_on("py-setuptools@61:", when="@0.5:", type="build")
+    depends_on("py-setuptools@42:", type="build")
+    depends_on("py-einops@0.3:", type=("build", "run"))
+    depends_on("py-fiona@1.8.19:", when="@0.5:", type=("build", "run"))
+    depends_on("py-fiona@1.8:", when="@0.3:", type=("build", "run"))
+    depends_on("py-fiona@1.5:", type=("build", "run"))
+    # Only part of lightning[pytorch-extra] we actually require
+    depends_on("py-jsonargparse@4.18:+signatures", when="@0.5:", type=("build", "run"))
+    depends_on("py-kornia@0.6.9:", when="@0.5:", type=("build", "run"))
+    depends_on("py-kornia@0.6.5:", when="@0.4.1:", type=("build", "run"))
     depends_on("py-kornia@0.6.5:0.6.9", when="@0.4.0", type=("build", "run"))
     depends_on("py-kornia@0.6.4:0.6.9", when="@0.3", type=("build", "run"))
     depends_on("py-kornia@0.5.11:0.6.9", when="@0.2", type=("build", "run"))
     depends_on("py-kornia@0.5.4:0.6.9", when="@0.1", type=("build", "run"))
```

Comment:
`py-kornia` is the only dependency that still has its upper bounds, I guess because there are known issues. What do you think about adding a comment so that we don't think it is an oversight in the future?